File: Examples/ObjcDemo/RixSDKPodDemo/RixEngineObjcDemo/UI/Rixengine/AlxRewardVideoVC.m
#import "AlxRewardVideoVC.h"
#import <AlxAds/AlxAds-Swift.h>
#import "AdConfig.h"

@interface AlxRewardVideoVC () <AlxRewardVideoAdDelegate>

@property (nonatomic, strong) AlxRewardVideoAd *rewardAd;
@property (nonatomic, strong) UILabel *label;
@property (nonatomic, assign) BOOL isLoading;
@property (nonatomic, copy) NSString *TAG;

@end

@implementation AlxRewardVideoVC

- (void)viewDidLoad {
    [super viewDidLoad];
    self.TAG = @"Alx-rewardVideo:";
    self.view.backgroundColor = [UIColor whiteColor];
    self.navigationItem.title = NSLocalizedString(@"rewardVideo_ad", @"");

    UIButton *bnLoad = [self createButtonWithTitle:NSLocalizedString(@"load_ad", @"") action:@selector(loadAd)];
    [self.view addSubview:bnLoad];

    UIButton *bnShow = [self createButtonWithTitle:NSLocalizedString(@"show_ad", @"") action:@selector(showAd)];
    [self.view addSubview:bnShow];

    self.label = [self createLabel];
    [self.view addSubview:self.label];

    [NSLayoutConstraint activateConstraints:@[
        [bnLoad.leadingAnchor constraintEqualToAnchor:self.view.leadingAnchor],
        [bnLoad.trailingAnchor constraintEqualToAnchor:self.view.trailingAnchor],
        [bnLoad.heightAnchor constraintEqualToConstant:50],
        [bnLoad.topAnchor constraintEqualToAnchor:self.view.safeAreaLayoutGuide.topAnchor],

        [bnShow.leadingAnchor constraintEqualToAnchor:self.view.leadingAnchor],
        [bnShow.trailingAnchor constraintEqualToAnchor:self.view.trailingAnchor],
        [bnShow.heightAnchor constraintEqualToConstant:50],
        [bnShow.topAnchor constraintEqualToAnchor:bnLoad.bottomAnchor constant:20],

        [self.label.leadingAnchor constraintEqualToAnchor:self.view.leadingAnchor],
        [self.label.trailingAnchor constraintEqualToAnchor:self.view.trailingAnchor],
        [self.label.heightAnchor constraintEqualToConstant:50],
        [self.label.topAnchor constraintEqualToAnchor:bnShow.bottomAnchor constant:20],
    ]];

    [self createAd];
}

- (void)createAd {
    self.rewardAd = [[AlxRewardVideoAd alloc] init];
}

- (void)loadAd {
    if (self.isLoading) {
        return;
    }
    self.isLoading = YES;
    self.label.text = NSLocalizedString(@"loading", @"");

    self.rewardAd.delegate = self;
    [self.rewardAd loadAdWithAdUnitId:[AdConfig Alx_Reward_Video_Ad_Id]];
}

- (void)showAd {
    if ([self.rewardAd isReady]) {
        [self.rewardAd showAdWithPresent:self];
    }
}

#pragma mark - AlxRewardVideoAdDelegate

- (void)rewardVideoAdLoad:(AlxRewardVideoAd *)ad {
    NSLog(@"%@ rewardVideoAdLoad", self.TAG);
    self.isLoading = NO;
    self.label.text = NSLocalizedString(@"load_success", @"");
}

- (void)rewardVideoAdFailToLoad:(AlxRewardVideoAd *)ad didFailWithError:(NSError *)error {
    NSString *msg = [NSString stringWithFormat:@"%ld: %@", (long)error.code, error.localizedDescription];
    NSLog(@"%@ rewardVideoAdFailToLoad: %@", self.TAG, msg);
    self.isLoading = NO;
    self.label.text = [NSString stringWithFormat:NSLocalizedString(@"load_failed", @""), msg];
}

- (void)rewardVideoAdImpression:(AlxRewardVideoAd *)ad {
    NSLog(@"%@ rewardVideoAdImpression", self.TAG);
}

- (void)rewardVideoAdClick:(AlxRewardVideoAd *)ad {
    NSLog(@"%@ rewardVideoAdClick", self.TAG);
}

- (void)rewardVideoAdClose:(AlxRewardVideoAd *)ad {
    NSLog(@"%@ rewardVideoAdClose", self.TAG);
}

- (void)rewardVideoAdPlayStart:(AlxRewardVideoAd *)ad {
    NSLog(@"%@ rewardVideoAdPlayStart", self.TAG);
}

- (void)rewardVideoAdPlayEnd:(AlxRewardVideoAd *)ad {
    NSLog(@"%@ rewardVideoAdPlayEnd", self.TAG);
}

- (void)rewardVideoAdReward:(AlxRewardVideoAd *)ad {
    NSLog(@"%@ rewardVideoAdReward", self.TAG);
}

- (void)rewardVideoAdPlayFail:(AlxRewardVideoAd *)ad didFailWithError:(NSError *)error {
    NSLog(@"%@ rewardVideoAdPlayFail %@", self.TAG, error.localizedDescription);
}

@end

File: AlxAdsDemo/UI/Rixengine/AlxRewardVideoVC.swift
//
//  AlxRewardVideoVC.swift
//  AlxDemo
//
//  Created by liu weile on 2025/3/31.
//

import UIKit
import AlxAds

class AlxRewardVideoVC: BaseUIViewController {
    
    private let TAG = "Alx-rewardVideo:"

    private var rewardAd:AlxRewardVideoAd!
    
    private var label:UILabel!
    
    private var isLoading:Bool=false
    

    override func viewDidLoad() {
        super.viewDidLoad()
        view.backgroundColor = .white
        navigationItem.title=NSLocalizedString("rewardVideo_ad", comment: "")
        
        let bnLoad=createButton(title: NSLocalizedString("load_ad", comment: ""),  action: #selector(loadAd))
        view.addSubview(bnLoad)
        
        let bnShow=createButton(title: NSLocalizedString("show_ad", comment: ""),  action: #selector(showAd))
        view.addSubview(bnShow)
        
        label=createLabel()
        view.addSubview(label)
        
        NSLayoutConstraint.activate([
            bnLoad.leftAnchor.constraint(equalTo: view.leftAnchor),
            bnLoad.rightAnchor.constraint(equalTo: view.rightAnchor),
            bnLoad.widthAnchor.constraint(equalTo: view.widthAnchor),
            bnLoad.heightAnchor.constraint(equalToConstant: 50),
            bnLoad.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor),
            
            bnShow.leftAnchor.constraint(equalTo: view.leftAnchor),
            bnShow.rightAnchor.constraint(equalTo: view.rightAnchor),
            bnShow.widthAnchor.constraint(equalTo: view.widthAnchor),
            bnShow.heightAnchor.constraint(equalToConstant: 50),
            bnShow.topAnchor.constraint(equalTo: bnLoad.bottomAnchor, constant: 20),
            
            label.leftAnchor.constraint(equalTo: view.leftAnchor),
            label.rightAnchor.constraint(equalTo: view.rightAnchor),
            label.widthAnchor.constraint(equalTo: view.widthAnchor),
            label.heightAnchor.constraint(equalToConstant: 50),
            label.topAnchor.constraint(equalTo: bnShow.bottomAnchor, constant: 20),
           
        ])
        
        createAd()
    }
    
    private func createAd(){
        rewardAd=AlxRewardVideoAd()
    }
    
    @objc func loadAd(){
        if isLoading {
            return
        }
        isLoading=true
        label.text=NSLocalizedString("loading", comment: "")
        
        rewardAd.delegate = self
        rewardAd.loadAd(adUnitId: AdConfig.Alx_Reward_Video_Ad_Id)
    }
    
    @objc func showAd(){
        if rewardAd.isReady() {
            rewardAd.showAd(present: self)
        }
    }

}

extension AlxRewardVideoVC:AlxRewardVideoAdDelegate{
    
    func rewardVideoAdLoad(_ ad: AlxRewardVideoAd) {
        print("\(TAG) rewardVideoAdLoad")
        self.isLoading=false
        self.label.text=NSLocalizedString("load_success", comment: "")
    }
    
    func rewardVideoAdFailToLoad(_ ad: AlxRewardVideoAd, didFailWithError error: Error) {
        let error1=error as NSError
        let msg="\(error1.code): \(error1.localizedDescription)"
        print("\(TAG) rewardVideoAdFailToLoad: \(msg)")
        
        self.isLoading=false
        self.label.text=String(format: NSLocalizedString("load_failed", comment: ""), msg)
    }
    
    func rewardVideoAdImpression(_ ad: AlxRewardVideoAd) {
        print("\(TAG) rewardVideoAdImpression")
    }
    
    func rewardVideoAdClick(_ ad: AlxRewardVideoAd) {
        print("\(TAG) rewardVideoAdClick")
    }
    
    func rewardVideoAdClose(_ ad: AlxRewardVideoAd) {
        print("\(TAG) rewardVideoAdClose")
    }
    
    func rewardVideoAdPlayStart(_ ad:AlxRewardVideoAd) {
        print("\(TAG) rewardVideoAdPlayStart")
    }
    
    func rewardVideoAdPlayEnd(_ ad: AlxRewardVideoAd) {
        print("\(TAG) rewardVideoAdPlayEnd")
    }
    
    func rewardVideoAdReward(_ ad: AlxRewardVideoAd) {
        print("\(TAG) rewardVideoAdReward")
    }
    
    func rewardVideoAdPlayFail(_ ad: AlxRewardVideoAd, didFailWithError error: Error) {
        print("\(TAG) rewardVideoAdPlayFail \(error.localizedDescription)")
    }
    
        
    
}


